dynamic_programming/fibonacci_bottom_up.cpp

```diff
@@ -1,6 +1,36 @@
-#include <iostream>
-using namespace std;
+/**
+ * @file
+ * @brief Implementation to find n th fibonacci_number
+ * (https://en.wikipedia.org/wiki/Fibonacci_number)
+ *
+ * @details
+ * The Fibonacci sequence is a series of numbers where each number is the
+ * sum of the two preceding ones, starting with 0 and 1.
+ *
+ * ### Algorithm
+ * The idea of the algorithm is to find the ith fibonacci number.
+ * DP state : F(i) = ith fibonacci number
+ * DP state transition :
+ *      F(i) = F(i-1) + F(i-2)
+ * DP base case :
+ *      F(0) = 0
+ *      F(1) = 1
+ * Space reduction :
+ * for i the fibonacci store only i-1 th and i-2 th state.
+ */
+#include <cassert>   /// for assert
+#include <iostream>  /// for IO operations
+/**
+ * @namespace dynamic_programming
+ * @brief Dynamic Programming algorithms
+ */
 int fib(int n) {
+    if (n < 0) {
+        return -1;
+    }
+    if (n == 0) {
+        return 0;
+    }
     int res[3];
```

Comment:
Let's use a C++-style array (`std::vector`) instead. This will also fix the CI warnings.
The `vector` library/header must be added in order for this to work.
```suggestion
    std::vector<int> res(3);
```